onnx/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/OnnxInferenceModel.kt

```diff
@@ -320,6 +322,40 @@ public open class OnnxInferenceModel private constructor(private val modelSource
             return inputTensor
         }
 
+        private fun checkTensorMatchesInputShape(data: FloatArray, inputShape: LongArray) {
+            val numOfElements = inputShape.reduce { acc, dim -> acc * dim }.toInt()
+
+            if (data.size == numOfElements) return
+
+            if (inputShape.matchNHWC(C = 3) || inputShape.matchNCHW(C = 3)) {
+                val (height, width) = when {
```

Comment:
Do we really need to know `width` and `height` and the layout here? We can simplify this condition as something like:
```kotlin
inputShape.size == 4 && inputShape[0] == 1L && 
        (inputShape[1] == 3L || inputShape[3] == 3L) &&
                (data.size * 3 == numOfElements)
```

Reply:
agree!